gateway/oauth_manager_test.go

```diff
@@ -1357,3 +1361,135 @@ func TestPurgeOAuthClientTokensEvent(t *testing.T) {
 	assertTokensLen(t, storageManager, storageKey2, 0)
 
 }
+
+func TestPurgeOAuthClientTokensInBackground(t *testing.T) {
+	conf := func(globalConf *config.Config) {
+		// set tokens to be expired after 1 second
+		globalConf.OauthTokenExpire = 1
+		// cleanup tokens older than 2 seconds
+		globalConf.OauthTokenExpiredRetainPeriod = 2
+	}
+
+	ts := StartTest(conf)
+	defer ts.Close()
+
+	spec := ts.LoadTestOAuthSpec()
+
+	clientID1, clientID2 := uuid.New(), uuid.New()
+
+	ts.createOAuthClientIDAndTokens(t, spec, clientID1)
+	ts.createOAuthClientIDAndTokens(t, spec, clientID2)
+	storageKey1, storageKey2 := fmt.Sprintf("%s%s", prefixClientTokens, clientID1),
+		fmt.Sprintf("%s%s", prefixClientTokens, clientID2)
+
+	storageManager := ts.Gw.getGlobalMDCBStorageHandler(generateOAuthPrefix(spec.APIID), false)
+	storageManager.Connect()
+
+	assertTokensLen(t, storageManager, storageKey1, 3)
+	assertTokensLen(t, storageManager, storageKey2, 3)
+
+	time.Sleep(time.Second * 3)
```

Comment:
This is still a read flag, we shouldn't sleep this much

Reply:
I understand the significant delay introduced here, but we could set `oauth_token_expire` and `oauth_token_expire_retain_period` to be 1, which can reduce it to 2 seconds and not less than that, since they're values in seconds